src/eigen.jl

```diff
@@ -221,188 +221,160 @@ end
     return Eigen(vals,vecs)
 end
 
-# A small part of the code in the following method was inspired by works of David
-# Eberly, Geometric Tools LLC, in code released under the Boost Software
-# License (included at the end of this file).
+# Port of https://www.geometrictools.com/GTEngine/Include/Mathematics/GteSymmetricEigensolver3x3.h
+# released by David Eberly, Geometric Tools, Redmond WA 98052
+# under the Boost Software License, Version 1.0 (included at the end of this file)
+# TODO ensure right-handedness of the eigenvalue matrix
 # TODO extend the method to complex hermitian
-@inline function _eig(::Size{(3,3)}, A::LinearAlgebra.HermOrSym{T}, permute, scale) where {T <: Real}
-    S = arithmetic_closure(T)
-    Sreal = real(S)
-
-    @inbounds a11 = convert(Sreal, A.data[1])
-    @inbounds a22 = convert(Sreal, A.data[5])
-    @inbounds a33 = convert(Sreal, A.data[9])
-    if A.uplo == 'U'
-        @inbounds a12 = convert(S, A.data[4])
-        @inbounds a13 = convert(S, A.data[7])
-        @inbounds a23 = convert(S, A.data[8])
-    else
-        @inbounds a12 = conj(convert(S, A.data[2]))
-        @inbounds a13 = conj(convert(S, A.data[3]))
-        @inbounds a23 = conj(convert(S, A.data[6]))
-    end
-
-    p1 = abs2(a12) + abs2(a13) + abs2(a23)
-    if (p1 == 0)
-        # Matrix is diagonal
-        v1 = SVector(one(S),  zero(S), zero(S))
-        v2 = SVector(zero(S), one(S),  zero(S))
-        v3 = SVector(zero(S), zero(S), one(S) )
-
-        if a11 < a22
-            if a22 < a33
-                return Eigen(SVector((a11, a22, a33)), hcat(v1,v2,v3))
-            elseif a33 < a11
-                return Eigen(SVector((a33, a11, a22)), hcat(v3,v1,v2))
-            else
-                return Eigen(SVector((a11, a33, a22)), hcat(v1,v3,v2))
-            end
-        else #a22 < a11
-            if a11 < a33
-                return Eigen(SVector((a22, a11, a33)), hcat(v2,v1,v3))
-            elseif a33 < a22
-                return Eigen(SVector((a33, a22, a11)), hcat(v3,v2,v1))
-            else
-                return Eigen(SVector((a22, a33, a11)), hcat(v2,v3,v1))
-            end
+@inline function StaticArrays._eig(::Size{(3,3)}, A::LinearAlgebra.HermOrSym{T}, permute, scale) where {T <: Real}
+    function converged(aggressive, bdiag0, bdiag1, bsuper)
+        if aggressive
+            bsuper == 0
+        else
+            summe = abs(bdiag0) + abs(bdiag1)
```

Comment:
`summe`?

Reply:
Was `sum` in the original and I'm always kinda anxious of shadowing Base verbs.
Fixed by a474f02.